src/controllers/keystore/keystore.ts

```diff
@@ -621,11 +607,17 @@ export class KeystoreController extends EventEmitter {
     this.emitUpdate()
   }
 
+  get hasPasswordSecret() {
+    return !!this.#keystoreSecrets.find((x) => x.id === 'password')
```

Comment:
Minor: you can use the `some` method instead of `find` to return a boolean directly without needing to use `!!`, like so:

```
return this.#keystoreSecrets.some((x) => x.id === 'password');
```